bentoml/handlers/tensorflow_tensor_handler.py

```diff
@@ -16,19 +16,171 @@
+
+    Raises:
+        BentoMLException: BentoML currently doesn't support Content-Type
     """
+    METHODS = (
+        PREDICT,
+        CLASSIFY,
+        REGRESS,
+    ) = (
+        "predict",
+        "classify",
+        "regress",
+    )
+
+    def __init__(self, method=PREDICT):
+        self.method = method
+
+    @property
+    def request_schema(self):
+        if self.method == self.PREDICT:
+            return {
+                "application/json": {
+                    "schema": {
+                        "type": "object",
+                        "properties": {
+                            "signature_name": {
+                                "type": "string",
+                                "default": None,
+                            },
+                            "instances": {
+                                "type": "array",
+                                "items": {
+                                    "type": "object",
+                                },
+                                "default": None,
+                            },
+                            "inputs": {
+                                "type": "object",
+                                "default": None,
+                            }
+                        },
+                    }
+                }
+            }
+        else:
+            raise NotImplementedError(f"method {self.method} is not implemented")
+
+    def _handle_raw_str(self, raw_str, output_format, func):
+        parsed_json = json.loads(raw_str)
+        if parsed_json.get("instances") is not None:
+            instances = parsed_json.get("instances")
+            instances = decode_b64_if_needed(instances)
+            parsed_tensor = tf.constant(instances)
+            result = func(parsed_tensor)
+            if isinstance(result, tf.Tensor):
+                result = result.numpy().tolist()
+            if isinstance(result, np.ndarray):
+                result = result.tolist()
+
+        elif parsed_json.get("inputs"):
+            raise NotImplementedError("column format 'inputs' is not implemented")
+
+        if output_format == "json":
+            result_object = {"predictions": result}
+            result_str = get_output_str(result_object, output_format)
+        elif output_format == "str":
+            result_str = get_output_str(result, output_format)
+
+        return result_str
 
     def handle_request(self, request, func):
-        raise NotImplementedError
+        """Handle http request that has jsonlized tensorflow tensor. It will convert it
+        into a tf tensor for the function to consume.
+
+        Args:
+            request: incoming request object.
+            func: function that will take ndarray as its arg.
+        Return:
+            response object
+        """
+        output_format = request.headers.get("output", "json")
+        if output_format not in {"json", "str"}:
+            return make_response(
```

Comment:
use `raise BadInput` instead?  we recently made some changes to how we handle this type of error. Similarily below in L157